pkg/resource/v1/chartconfig/current.go

```diff
@@ -2,8 +2,45 @@ package chartconfig
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GetCurrentState returns the ChartConfig resources present in the guest
+// cluster.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for chartconfigs in the guest cluster")
+
+	clusterConfig, err := prepareClusterConfig(r.baseClusterConfig, clusterGuestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	g8sClient, err := r.guestClusterService.GetG8sClient(ctx, clusterConfig.ClusterID, clusterConfig.Domain.API)
```

Comment:
I think instead of `guestClusterService` it would be good enough to call it just `guest`. Within the scope of the chartconfig resource this should work. But this may just be personal. 

Further, for creation of objects which provide functionality the constructors are usually called `New*`. I find that better than the `Get*` naming since we do not fetch the client.

Reply:
Good points. I'll rename to `guest`. Same with `NewG8sClient` I had it as `G8sClient` before but that is better.

Reply:
Done.